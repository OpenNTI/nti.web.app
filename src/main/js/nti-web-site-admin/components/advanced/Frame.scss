.advanced-admin-tools-view {
	margin-top: 40px;
}
