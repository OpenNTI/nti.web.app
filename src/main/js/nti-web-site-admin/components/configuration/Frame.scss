.advanced-admin-tools-view {
	margin-top: 20px;
}
