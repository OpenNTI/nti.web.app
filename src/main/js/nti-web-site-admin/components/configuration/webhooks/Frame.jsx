import { scoped } from '@nti/lib-locale';
import { DataContext } from '@nti/web-core/data';
import { LinkTo } from '@nti/web-routing';
import { Box, Icons, Text } from '@nti/web-core';

import { Store as Store } from './Store';

const t = scoped('nti-web-app.admin.config.Webhooks', {
	title: 'Webhook Subscriptions',
	back: 'Back to Subscriptions',
});

const Header = styled.div`
	padding: 1rem 0;

	& a {
		color: var(--primary-grey);
		display: flex;
		flex-direction: row;
		align-items: center;
	}

	& i {
		font-size: 0.875rem;
		font-weight: 700;
		width: 1rem;
	}
`;

export const Frame = ({ children, ...props }) => {
	return (
		<Box p="lg" sh="sm">
			{children && (
				<Header>
					<LinkTo.Name name="site-admins.config.webhooks">
						<Icons.Chevron.Left />
						<span>{t('back')}</span>
					</LinkTo.Name>
				</Header>
			)}
			<Text as="h1">{t('title')}</Text>
			<DataContext store={Store.useStore()} fallback={<div />}>
				{children}
			</DataContext>
		</Box>
	);
};
