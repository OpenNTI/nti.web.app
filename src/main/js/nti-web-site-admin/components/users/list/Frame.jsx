import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from '@nti/web-routing';// eslint-disable-line
import {Layouts, FixedElement} from '@nti/web-commons';

import NavBar from './nav-bar';

export default class SiteAdminUserListFrame extends React.Component {
	static propTypes = {
		children: PropTypes.node
	}

	render () {
		const {children} = this.props;

		return (
			<div className="site-admin-user-list-frame">
				<Layouts.NavContent.Container>
					<Layouts.NavContent.Nav className="nav-bar">
						{/* {this.renderHeader()} */}
						<FixedElement>
							<NavBar/>
						</FixedElement>
					</Layouts.NavContent.Nav>
					<Layouts.NavContent.Content className="content">
						{React.Children.map(children, (item) => {
							return React.cloneElement(item);
						})}
					</Layouts.NavContent.Content>
				</Layouts.NavContent.Container>
			</div>
		);
	}
}
