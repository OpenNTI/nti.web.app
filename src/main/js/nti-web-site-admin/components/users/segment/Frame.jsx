import { scoped } from '@nti/lib-locale';
import { DataContext } from '@nti/web-core/data';
import { Icons, ErrorMessage } from '@nti/web-core';
import { LinkTo } from '@nti/web-routing';

import { SegmentStore } from './Store';

const t = scoped('nti-site-admin.components.users.segment.Frame', {
	back: 'Back to Segments',
});

const Header = styled.div`
	padding: 1rem 0;

	& a {
		color: var(--primary-grey);
		display: flex;
		flex-direction: row;
		align-items: center;
	}

	& i {
		font-size: 0.875rem;
		font-weight: 700;
		width: 1rem;
	}
`;

export function SegmentFrame({ children, segmentID }) {
	const store = SegmentStore.useStore({ segmentID });

	return (
		<DataContext store={store} fallback={<div />} error={<ErrorMessage />}>
			<Header>
				<LinkTo.Name name="site-admins.user.user-segments">
					<Icons.Chevron.Left />
					<span>{t('back')}</span>
				</LinkTo.Name>
			</Header>
			{children}
		</DataContext>
	);
}
