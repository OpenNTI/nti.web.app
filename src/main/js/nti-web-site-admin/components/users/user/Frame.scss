@import '~nti-style-common';

.site-admin-user-view {
	min-height: 200px;

	.header {
		font-size: 1rem;
		padding: 1rem 0;

		a {
			text-decoration: none;
			color: $primary-grey;
			display: flex;
			align-items: center;
		}

		i {
			font-size: 0.875rem;
			font-weight: 700;
			width: 1rem;
		}
	}

	.content {
		padding-top: rem-calc(54px);//the height of the header
	}
}
