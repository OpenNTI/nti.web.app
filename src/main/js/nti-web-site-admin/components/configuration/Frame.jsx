import './Frame.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from '@nti/web-routing';// eslint-disable-line
import {Layouts, FixedElement} from '@nti/web-commons';


import NavBar from './nav-bar';

export default class SiteAdminAdvancedView extends React.Component {
	static propTypes = {
		children: PropTypes.node
	}

	render () {
		return (
			<div className="advanced-admin-tools-view">
				{this.renderLayout()}
			</div>
		);

	}

	renderLayout () {
		const {children} = this.props;

		return (
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					<FixedElement>
						<NavBar/>
					</FixedElement>
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, (item) => {
						return React.cloneElement(item, {});
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		);
	}
}
