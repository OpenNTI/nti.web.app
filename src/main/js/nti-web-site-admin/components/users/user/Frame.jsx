import './Frame.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from '@nti/web-routing';
import {decorate} from '@nti/lib-commons';
import {Loading, Layouts} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';

import Store from './Store';
import NavBar from './nav-bar';

const DEFAULT_TEXT = {
	back: 'Back to People'
};

const t = scoped('nti-site-admin.users.user.Frame', DEFAULT_TEXT);

class SiteAdminUserView extends React.Component {
	static propTypes = {
		userID: PropTypes.string,

		user: PropTypes.object,
		hasBooks: PropTypes.bool,
		hasCourses: PropTypes.bool,
		loading: PropTypes.bool,
		store: PropTypes.object,

		children: PropTypes.node
	}


	get store () {
		return this.props.store;
	}


	componentDidUpdate (oldProps) {
		const {userID: newID} = this.props;
		const {userID: oldID} = oldProps;

		if (newID !== oldID) {
			this.store.loadUser(newID);
		}
	}


	componentDidMount () {
		const {userID} = this.props;

		this.store.loadUser(userID);
	}


	componentWillUnmount () {
		const {user} = this.props;

		this.store.unloadUser(user);
	}

	render () {
		const {loading} = this.props;

		return (
			<div className="site-admin-user-view">
				{loading && this.renderHeader()}
				{loading && (<Loading.Mask />)}
				{!loading && this.renderUser()}
			</div>
		);

	}

	renderUser () {
		const {user, hasBooks, hasCourses, children} = this.props;

		if (!user) {
			return null;
		}

		return (
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					{this.renderHeader()}
					<NavBar user={user} hasBooks={hasBooks} hasCourses={hasCourses}/>
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, (item) => {
						return React.cloneElement(item, {user});
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		);
	}


	renderHeader () {
		return (
			<div className="header">
				<LinkTo.Path to="../..">
					<i className="icon-chevron-left" />
					<span>{t('back')}</span>
				</LinkTo.Path>
			</div>
		);
	}
}


export default decorate(SiteAdminUserView, [
	Store.connect({user: 'user', loading: 'loading', hasBooks: 'hasBooks', hasCourses: 'hasCourses'})
]);
