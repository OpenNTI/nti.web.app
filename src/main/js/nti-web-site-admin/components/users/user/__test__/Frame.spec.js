import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import Frame from '../Frame';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = () => {
	return {
		resolveEntity: () => {
			return {
				hasLink: () => true,
				fetchLink: () => {
					return {};
				},
				avatarURL: 'someAvatarURL',
			};
		},
		getObject: () => {
			return {
				title: 'abc',
			};
		},
	};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe.skip('Site admin user frame test', () => {
	beforeEach(() => onBefore());
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<Frame userID="user1" />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
