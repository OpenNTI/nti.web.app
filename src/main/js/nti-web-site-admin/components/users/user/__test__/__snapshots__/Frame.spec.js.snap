// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user frame test Basic render test 1`] = `
<div
  className="site-admin-user-view"
>
  <div
    className="nav-content-container"
  >
    <div
      className="nav-content-nav nav-bar"
    >
      <div
        className="header"
      >
        <div
          className="nti-link-to-path"
          to="../.."
        >
          <i
            className="icon-chevron-left"
          />
          <span>
            Back to People
          </span>
        </div>
      </div>
      <div
        className="site-admin-card site-admin-user-nav-bar"
      >
        <div
          className="site-admin-user-identity"
        >
          <img
            alt="Avatar for undefined"
            className="avatar avatar-color-6"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          />
          <span
            className="username"
            rel="author"
          />
          <div
            className="actions"
          >
            <span
              className="email"
            />
            <div
              className="nti-link-to-path profile"
              to="#"
            >
              <i
                className="icon-goto"
              />
              <span>
                Profile
              </span>
            </div>
          </div>
        </div>
        <ul
          className="unadorned-list site-admin-tabs"
        >
          <li>
            <div
              className="nti-link-to-path"
              exact={true}
              to="./"
            >
              Overview
            </div>
          </li>
          <li />
          <li>
            <div
              className="nti-link-to-path"
              to="./courses"
            >
              NextThought.view.library.View.course
            </div>
          </li>
          <li>
            <div
              className="nti-link-to-path"
              to="./transcript"
            >
              Transcript
            </div>
          </li>
          <li>
            <div
              className="nti-link-to-path"
              to="./reports"
            >
              Reports
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="nav-content-content content"
    />
  </div>
</div>
`;
